foreman/data_refinery_foreman/surveyor/test_end_to_end.py

```diff
@@ -38,10 +39,99 @@
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680046/suppl/GSM2680046_C4_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680046_C4_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680047/suppl/GSM2680047_C5_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680047_C5_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680048/suppl/GSM2680048_WD1_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680048_WD1_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680049/suppl/GSM2680049_WD2_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680049_WD2_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680050/suppl/GSM2680050_WD3_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680050_WD3_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680051/suppl/GSM2680051_WD4_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680051_WD4_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680052/suppl/GSM2680052_WD6_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680052_WD6_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680053/suppl/GSM2680053_WCu1_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680053_WCu1_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680054/suppl/GSM2680054_WCu2_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680054_WCu2_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680055/suppl/GSM2680055_WCu3_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680055_WCu3_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680056/suppl/GSM2680056_WCu7_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680056_WCu7_Cu.CEL.gz",  # noqa
+    "ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2680nnn/GSM2680057/suppl/GSM2680057_WCu8_Cu.CEL.gz": "https://data-refinery-test-assets.s3.amazonaws.com/end_to_end_downloads/GSM2680057_WCu8_Cu.CEL.gz",  # noqa
+}
+
+
+def build_surveyor_init_mock(source_type):
+    if source_type == "ARRAY_EXPRESS":
+        original_surveyor = ORIGINAL_ARRAY_EXPRESS_SURVEYOR
+    if source_type == "SRA":
+        original_surveyor = ORIGINAL_SRA_SURVEYOR
+    if source_type == "TRANSCRIPTOME_INDEX":
+        original_surveyor = ORIGINAL_TRANSCRIPTOME_SURVEYOR
+    if source_type == "GEO":
+        original_surveyor = ORIGINAL_GEO_SURVEYOR
```

Comment:
Should we throw an error if an unknown `source_type` is provided?

Reply:
An error will be thrown, because `original_surveyor` will be undefined. Do you think that won't be specific enough? This is test code so presumably that would only happen if a developer was modifying the tests and wrote code that expected the source_type to be in here and it wasn't.